jwst/assign_wcs/niriss.py

```diff
@@ -164,8 +171,151 @@ def imaging_distortion(input_model, reference_files):
+            uberimage to that of the dispersed image.  For the Vanilla
+            Pipeline we assume that the pointing information in the file
+            headers is sufficient.  This will be strictly true if all images
+            were obtained in a single visit (same guide stars).
+    Step 2: Record source information for each object in the catalog: position
+            (RA and Dec), shape (A_IMAGE, B_IMAGE, THETA_IMAGE), and all
+            available magnitudes.
+    Step 3: Compute the trace and wavelength solutions for each object in the
+            catalog and for each spectral order.  Record this information.
+    Step 4: Compute the WIDTH of each spectral subwindow, which may be fixed or
+            variable (see discussion of optimal extraction, below).  Record
+            this information.
+    Step 4: Record the MMAG_EXTRACT for each object and spectral order.
+
+    Source catalog use moved to extract_2d
+    """
+
+    # The input is the grism image
+    if not isinstance(input_model, ImageModel):
+        raise TypeError('The input data model must be an ImageModel.')
+
+    # make sure this is a grism image
+    if "NIS_WFSS" not in input_model.meta.exposure.type:
+            raise TypeError('The input exposure is not NIRISS grism')
+
+    # Create the empty detector as a 2D coordinate frame in pixel units
+    gdetector = cf.Frame2D(name='grism_detector',
+                           axes_order=(0, 1),
+                           unit=(u.pix, u.pix))
+
+    # translate the x,y detector-in to x,y detector out coordinates
+    # Get the disperser parameters which are defined as a model for each
+    # spectral order
+    with NIRISSGrismModel(reference_files['specwcs']) as f:
+        dispx = f.dispx
+        dispy = f.dispy
+        displ = f.displ
+        invdispl = f.invdispl
+        orders = f.orders
+        fwcpos_ref = f.fwcpos_ref
+
+    # sep the row and column grism models
+    if 'R' in input_model.instrument.pupil[-1]:
+        det2det = NIRISSForwardRowGrismDispersion(orders, displ, dispx, dispy, fwcpos_ref)
+    else:
+        det2det = NIRISSForwardColumnGrismDispersion(orders, displ, dispx, dispy, fwcpos_ref)
+
+    backward = NIRISSBackwardGrismDispersion(orders, invdispl, dispx, dispy, fwcpos_ref)
+    det2det.inverse = backward
+
+    # create the pipeline to construct a WCS object for the whole image
+    # which can translate ra,dec to image frame reference pixels
+    # it also needs to be part of the grism image wcs pipeline to
+    # go from detector to world coordinates. However, the grism image
+    # will be effectively translating pixel->world coordinates in a
+    # manner that gives you the originating pixels ra and dec, not the
+    # pure ra/dec on the sky from the pointing wcs.
+
+    # use the imaging_distortion reference file here
+    img_reference = deepcopy(reference_files)
+    img_reference['distortion'] = reference_files['imaging_distortion']
```

Comment:
Isn't this just `reference_files['distortion']` instead of `reference_files['imaging_distortion']` or was this set somewhere else and I missed it?

Reply:
yes, it should just be distortion. I missed changing this when the teams agreed to redeliver the imaging distortion refs